### PR TITLE
Preserve Data

### PR DESCRIPTION
**Pull request by tafaramafemba**
Hello 😁😁😁😁👋👋👋

In this pull request, we have managed to do the following 🔨🔨🔨:

1. On exit from the app all the data is saved (books, people, and rentals) into JSON files. One per type of data stored and named accordingly (e.g. rentals.json) ✔
2. Load all data in the app on start up ✔
3. Handles the case when one or more of the files are missing. ✔

Enjoy the review 💃💃💃

### CODE REVIEW

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a group project, that has commits from one partner please invite your partner to make commits to this pull request before submitting it for review.

- [x] Please kindly check your readme file and update it using the [Microverse Readme Template](https://github.com/microverseinc/readme-template)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: data/users.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"instance":"Student","id":835,"age":48,"name":"Thomas","classroom":"8 green"}]
```
Comment (brytebee) [inline data/users.json]: - [x] **Required:** The JSON is parsed in the wrong order and displayed wrongly while I tried to create a rental.

image

- Also when I attempt to list people.

image

Kindly fix this!

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi Team👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- On exit from the app the data is being saved.✔️
- On startup of the app load all data from files.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data/users.json (hunk 1 above)
Comment (tafaramafemba) [reply data/users.json]: @brytebee please kindly assist. When I add the file, it adds in one order and when I list the contents, they are in that order. But if I close the app and start it again, the order changes

Comment (tafaramafemba) [reply data/users.json]: basically when I add data and view it, it  views correctly. But if I close and start the app again, the order changes

**Review by brytebee — APPROVED** [review]
Hi Team👋,

Great job on implementing the suggested changes.

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 




Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**